orca-keel/src/main/kotlin/com/netflix/spinnaker/orca/keel/task/ImportDeliveryConfigTask.kt

```diff
@@ -61,23 +61,19 @@ constructor(
 
       TaskResult.builder(ExecutionStatus.SUCCEEDED).context(emptyMap<String, Any?>()).build()
     } catch (e: RetrofitError) {
-      handleFailure(e, context)
+      handleRetryableFailures(e, context)
     } catch (e: Exception) {
       log.error("Unexpected exception while executing {}, aborting.", javaClass.simpleName, e)
       buildError(e.message)
     }
   }
 
-  private fun processDeliveryConfigLocation(trigger: Trigger, context: PublishDeliveryConfigContext): String {
-    if (trigger is GitTrigger) {
-      // if the pipeline has a git trigger, infer what context we can from the trigger
-      // FIXME: GitTrigger props are non-null, but IIUC branch and hash would be mutually exclusive, so what to do here?
-      if (trigger.hash != null && context.ref == null) {
+  private fun processDeliveryConfigLocation(trigger: Trigger, context: ImportDeliveryConfigContext): String {
+    if (trigger is SourceCodeTrigger) {
```

Comment:
Because of the serialization back-and-forth behavior I mentioned in the description, in the end this wasn't strictly necessary, but I thought it conveyed more clearly that it doesn't _have_ to be a `GitTrigger`, specifically.

Reply:
I like